prow/config/jobs.go

```diff
@@ -135,6 +135,8 @@ type Presubmit struct {
 
 	Reporter
 
+	JenkinsSpec
```

Comment:
Why doesn't this apply to periodic jobs?

Reply:
I'm also not sure we should be embedding this `JenkinsSpec` struct instead of making it a normal, named field like the `PodSpec` and `BuildSpec`. As is, it is unclear that `github_based_project: true` only applies to jenkins because it is embedded at the root of `Presubmit`.

Reply:
```

// Presubmit runs on PRs.
type Presubmit struct {
	JobBase

...
	GithubBasedProject bool `json:"github_based_project"` 
...
}
```
Do you mean kind of like this?

Reply:
No I'm suggesting that we don't embed the `JenkinsSpec` struct and instead make it a normal field. Right now we have:
```go
// Presubmit runs on PRs.
type Presubmit struct {
	JobBase
        ...
	JenkinsSpec
}
```
I'm suggesting we switch to:
```go
// Presubmit runs on PRs.
type Presubmit struct {
	JobBase
        ...
	JenkinsSpec `json:"jenkins_spec,omitempty"`
}
```

This is what we do in `JobBase` for the other types of Specs and it ensures that the member fields are scoped to the proper type of Spec for the agent: https://github.com/kubernetes/test-infra/blob/46a2fa5e5253f703e6be4852065e39850a9b321b/prow/config/jobs.go#L103-L106